invoices_service.go

```diff
@@ -160,3 +160,23 @@ func (s *invoicesImpl) MarkFailed(invoiceNumber int) (*Response, *Invoice, error
 
 	return resp, &dst, err
 }
+
+// Allows custom invoice amounts to be refunded and generates a refund invoice.
+// Full open amount refunds of invoices with an unsettled transaction will void
+// the transaction and generate a void invoice.
+// https://dev.recurly.com/docs/line-item-refunds
+func (s *invoicesImpl) RefundVoidOpenAmount(invoiceNumber int, amountInCents int, refundApplyOrder string) (*Response, *Invoice, error) {
+	action := fmt.Sprintf("invoices/%d/refund", invoiceNumber)
+	req, err := s.client.newRequest("POST", action, nil, Invoice{
+		AmountInCents:    amountInCents,
+		RefundApplyOrder: refundApplyOrder,
+	})
```

Comment:
If these two fields are only used to post this one specific endpoint, I'd prefer to create an inline struct and populate the values so we don't clutter the main `Invoice` struct.

```go
data := struct{
   XMLName xml.Name `xml:"invoice"`
   AmountInCents int `xml:"amount_in_cents,omitempty"`
   RefundApplyOrder string `xml:"refund_apply_order,omitempty"`
}{
   AmountInCents: amountInCents,
   RefundApplyOrder: refundApplyOrder,
}
req, err := s.client.newRequest("POST", action, nil, data)
```
Also, if both fields are empty, is recurly okay with receiving `<invoice></invoice>` or should we send no request body in that case? I know you have the sandbox API setup, may need to run some tests (I tried looking through their docs but it's not clear)